src/tools/illink/src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1410,10 +1410,14 @@ protected void MarkAssembly (AssemblyDefinition assembly, DependencyInfo reason)
 
 			MarkExportedTypesTarget.ProcessAssembly (assembly, Context);
 
-			if (ProcessReferencesStep.IsFullyPreservedAction (Annotations.GetAction (assembly))) {
-				if (!Context.TryGetCustomData ("DisableMarkingOfCopyAssemblies", out string? disableMarkingOfCopyAssembliesValue) ||
-					disableMarkingOfCopyAssembliesValue != "true")
+			var action = Annotations.GetAction (assembly);
+			if (ProcessReferencesStep.IsFullyPreservedAction (action)) {
+				if (action != AssemblyAction.Copy ||
+					!Context.TryGetCustomData ("DisableMarkingOfCopyAssemblies", out string? disableMarkingOfCopyAssembliesValue) ||
+					disableMarkingOfCopyAssembliesValue != "true") {
+					Console.Error.WriteLine($"MarkEntireAssembly {assembly.Name.Name}");
```

Comment:
The `Console.Error.WriteLine` is probably a left over from debugging... should be removed.